routing/dht/records.go

```diff
@@ -117,12 +124,19 @@ func (dht *IpfsDHT) verifyRecordLocally(r *pb.Record) error {
 // massive amplification attack on the dht. Use with care.
 func (dht *IpfsDHT) verifyRecordOnline(ctx context.Context, r *pb.Record) error {
 
-	// get the public key, search for it if necessary.
-	p := peer.ID(r.GetAuthor())
-	pk, err := dht.getPublicKeyOnline(ctx, p)
-	if err != nil {
-		return err
+	if len(r.Signature) > 0 {
+		// get the public key, search for it if necessary.
+		p := peer.ID(r.GetAuthor())
+		pk, err := dht.getPublicKeyOnline(ctx, p)
+		if err != nil {
+			return err
+		}
+
+		err = record.CheckRecordSig(r, pk)
+		if err != nil {
+			return err
+		}
```

Comment:
This cant be part of the validators since the validators have no way of fetching a public key.